vue-gtag.d.ts

```diff
@@ -38,7 +32,16 @@ declare module 'vue-gtag' {
 
   export interface VueGtag {
     pageview(pageView: PageView): void;
-    event(action: string, event: Event): void;
+
+    /**
+     * Send a Google Analytics Event.
+     *
+     * @see https://developers.google.com/analytics/devguides/collection/gtagjs/events
+     *
+     * @param action string that will appear as the event action in Google Analytics Event reports
+     * @param eventParams
+     */
+    event(action: string, eventParams?: Gtag.ControlParams | Gtag.EventParams | Gtag.CustomParams): void;
```

Comment:
no idea what these `Gtag.ControlParams | Gtag.EventParams | Gtag.CustomParams` are, could you explain it to me? I'm not a huge TS user :)

Reply:
Certainly! The vertical bar (`|`) denotes a [union type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#union-types), or a value can be one of several types.

These reference types that are declared in [`@types/gtag.js`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/gtag.js/index.d.ts):

<details>
<summary>(view type definitions)</summary>

```ts
interface ControlParams {
    groups?: string | string[];
    send_to?: string | string[];
    event_callback?: () => void;
    event_timeout?: number;
}
```

```ts
interface EventParams {
    checkout_option?: string;
    checkout_step?: number;
    content_id?: string;
    content_type?: string;
    coupon?: string;
    currency?: string;
    description?: string;
    fatal?: boolean;
    items?: Item[];
    method?: string;
    number?: string;
    promotions?: Promotion[];
    screen_name?: string;
    search_term?: string;
    shipping?: Currency;
    tax?: Currency;
    transaction_id?: string;
    value?: number;
    event_label?: string;
    event_category?: string;
}
```

```ts
interface CustomParams {
    [key: string]: any;
}
```
</details>

which, when unioned together, describe an object that could include any of the optional properties in `Gtag.ControlParams` and `Gtag.EventParams` – as well as _any_ other properties as defined by `Gtag.CustomParams`.

This aligns with [`gtag.js` documentation](https://developers.google.com/analytics/devguides/collection/gtagjs/events) where there are a set of conventional parameters (most notably `event_category`, `event_label`, and `value`), additional parameters corresponding to advanced functionality (i.e. [ecommerce](https://developers.google.com/analytics/devguides/collection/gtagjs/enhanced-ecommerce)), and the ability to pass _any_ parameters.

This function interface is similar to the [`event` command interface they defined over there](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6d993e42ec3406d5047ed0d46da167eb7eb39f80/types/gtag.js/index.d.ts#L12).

Reply:
I get what you mean using a third party typing definition, but I am not going to add a dependency on this package for typescript users. it's out of scope, so I guess I can't accept this changes you've made because depending on a dependency

Reply:
Yeah, no worries. I'll take a pass later today and remove the dependency. Thanks for the review!